addons/account/models/account_journal.py

```diff
@@ -484,6 +484,31 @@ def _check_auto_post_draft_entries(self):
     def _onchange_type(self):
         self.refund_sequence = self.type in ('sale', 'purchase')
 
+    @api.onchange('payment_sequence')
+    def _onchange_payment_sequence(self):
+        if self.payment_sequence:
+            existing_payment_moves = self.env['account.move'].search([('journal_id', '=', self._origin.id)]).filtered(lambda m: m.payment_id)
+            existing_stmt_line_moves = self.env['account.move'].search([('journal_id', '=', self._origin.id)]).filtered(lambda m: m.statement_line_id)
+            if not existing_stmt_line_moves and existing_payment_moves and existing_payment_moves[-1].sequence_prefix.startswith(self.code):
```

Comment:
I'm not sure I understand the last condition. Why is it needed ? And why `[-1]` ?

Reply:
It is needed because user could have resequenced the moves, then there will be no issue. And [-1] is to avoid a singleton error when having more than one existing move, as we're taking the last move from the journal to set the sequence, I thought it was better to take[-1] instead of [0] or any other

Reply:
Would it be possible for different existing_payment_moves to have different sequence_prefix ?

For instance by doing so:
- Move 1: BNK/1
- Move 2: BNK/2
- Move 3: BNK/3

Reset Move 3 to Draft, change its name to BBNNKK/3. Then the last condition would think that a resequence has happened, but not for all entries, therefore Move 1 and 2 will still raise "Another entry with the same name already exists." ?

Would that be possible ? In this case we check more than `[-1]` ?

Perhaps a test would be useful :)

Reply:
Indeed, I'll change the condition, thanks 🙂 .
I wanted to write a test but I didn't find a way to assert a warning message, any insight is welcome (and to be honest, I'm not convinced that a test to check a warning is absolutely necessary)